TestBatchEvict is unstable

## Flaky Test
TestBatchEvict

### Which jobs are failing
```bash
=== RUN   TestBatchEvict
[FAIL]  pkg/schedule/schedulers TestBatchEvict
    evict_leader_test.go:128: 
        	Error Trace:	/home/runner/work/pd/pd/pkg/schedule/schedulers/evict_leader_test.go:128
        	Error:      	"[evict-leader-scheduler {evict leader: from store 1 to one in [2 3], or to 2 (for compatibility)} (kind:leader, region:937(0, 0), createAt:2024-08-22 03:39:20.336444041 +0000 UTC m=+0.420246367, startAt:0001-01-01 00:00:00 +0000 UTC, currentStep:0, size:96, steps:[0:{transfer leader from store 1 to store 2}], timeout:[1m0s]) evict-leader-scheduler {evict leader: from store 1 to one in [2 3], or to 2 (for compatibility)} (kind:leader, region:236(0, 0), createAt:2024-08-22 03:39:20.336976915 +0000 UTC m=+0.420779241, startAt:0001-01-01 00:00:00 +0000 UTC, currentStep:0, size:96, steps:[0:{transfer leader from store 1 to store 2}], timeout:[1m0s])]" should have 3 item(s), but has 2
        	Test:       	TestBatchEvict
--- FAIL: TestBatchEvict (0.34s)
```


### CI link
https://github.com/tikv/pd/actions/runs/10501395410/job/29091312395?pr=8558


### Reason for failure (if possible)


### Anything else

--- a/pkg/schedule/schedulers/evict_leader_test.go
+++ b/pkg/schedule/schedulers/evict_leader_test.go
@@ -26,6 +26,7 @@ import (
 	"github.com/tikv/pd/pkg/schedule/types"
 	"github.com/tikv/pd/pkg/storage"
 	"github.com/tikv/pd/pkg/utils/operatorutil"
+	"github.com/tikv/pd/pkg/utils/testutil"
 )
 
 func TestEvictLeader(t *testing.T) {
@@ -115,7 +116,7 @@ func TestBatchEvict(t *testing.T) {
 	tc.AddLeaderStore(2, 0)
 	tc.AddLeaderStore(3, 0)
 	// the random might be the same, so we add 1000 regions to make sure the batch is full
-	for i := 1; i <= 1000; i++ {
+	for i := 1; i <= 10000; i++ {
 		tc.AddLeaderRegion(uint64(i), 1, 2, 3)
 	}
 	tc.AddLeaderRegion(6, 2, 1, 3)
@@ -124,9 +125,13 @@ func TestBatchEvict(t *testing.T) {
 	sl, err := CreateScheduler(types.EvictLeaderScheduler, oc, storage.NewStorageWithMemoryBackend(), ConfigSliceDecoder(types.EvictLeaderScheduler, []string{"1"}), func(string) error { return nil })
 	re.NoError(err)
 	re.True(sl.IsScheduleAllowed(tc))
-	ops, _ := sl.Schedule(tc, false)
-	re.Len(ops, 3)
+	testutil.Eventually(re, func() bool {
+		ops, _ := sl.Schedule(tc, false)
+		return len(ops) == 3
+	})
 	sl.(*evictLeaderScheduler).conf.Batch = 5
-	ops, _ = sl.Schedule(tc, false)
-	re.Len(ops, 5)
+	testutil.Eventually(re, func() bool {
+		ops, _ := sl.Schedule(tc, false)
+		return len(ops) == 5
+	})
 }

